packages/client/src/components/app/blocks/RowExplorer.svelte

```diff
@@ -0,0 +1,247 @@
+<script>
+  import Block from "components/Block.svelte"
+  import BlockComponent from "components/BlockComponent.svelte"
+  import { makePropSafe as safe } from "@budibase/string-templates"
+  import { generate } from "shortid"
+
+  export let dataSource
+  export let height
+
+  export let cardTitle
+  export let cardSubtitle
+  export let cardDescription
+  export let cardImageURL
+  export let cardSearchField
+
+  export let detailFields
+  export let detailTitle
+
+  const stateKey = generate()
+
+  let listDataProviderId
+  let listRepeaterId
+</script>
+
+<Block>
+  <BlockComponent
+    type="container"
+    props={{
+      direction: "row",
+    }}
+    styles={{
+      custom: `
+        height: ${height} !important;
+      `,
+    }}
+  >
+    <BlockComponent
+      type="dataprovider"
+      order={0}
+      bind:id={listDataProviderId}
+      props={{
+        dataSource,
+        paginate: true,
+        limit: 10,
+        filter: [
+          {
+            id: generate(),
```

Comment:
We can just hard code this to something (e.g. `0` or `1`) since these only need to be unique within the set of filters. By using a randomly generated ID here we're changing the filter prop on every update, meaning we're going to do a new search request on every render of the block which is unecessary.

Some other blocks actually leave out the `id` property altogether (probably an oversight) but thankfully they still work post ejection as they only contain a single filter and therefore an `undefined` ID is still unique! But we should still update the other blocks to also contain IDs at some stage (again not for now, just thought this was worth mentioning while I thought of it).

The same applies to all the usages of IDs inside filters and button actions here - we can update them all to just be hard coded and reduce prop changes.